EIPS/eip-2016.md

```diff
@@ -0,0 +1,241 @@
+status: Draft
+type: Standards Track
+category: Interface
+created: 2020-11-01
+requires: 155, 695
+---
+
+## Simple Summary
+
+An RPC method for adding Ethereum chains to wallet applications.
+
+## Abstract
+
+The `wallet_addEthereumChain` RPC method allows Ethereum applications ("dapps") to suggest chains to be added to the wallet application.
+The caller must specify a chain ID and some chain metadata.
+The wallet application may arbitrarily refuse or accept the request.
+`true` is returned if the chain was added, and an error otherwise.
+
+Important cautions for implementers of this method are included in the [Security Considerations](#security-considerations) section.
+
+## Motivation
+
+Although Mainnet is by far the most important, a dapp may require the user to interact with other Ethereum chains in order to function.
+Such a chain may or may not be supported by the user's wallet application.
+`wallet_addEthereumChain` enables dapps to request any necessary chains to be added to the user's wallet.
+This provides UX benefits for dapps and wallets that want to provide such functionality.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### `wallet_addEthereumChain`
+
+The method takes a `chainId` and some chain metadata, returning `true` if the chain was added to the wallet, and an error otherwise.
+
+The wallet **MAY** reject the request for any reason. The wallet **MUST** reject the request if the `chainId` is improperly formatted.
+
+> Note that this method makes **no** statement about whether the wallet should change the user's currently selected chain, if the wallet has a concept thereof.
+
+#### Parameters
+
+```typescript
+interface AddEthereumChainParams {
+  chainId: string;
+  blockExplorerUrl?: string;
+  chainName?: string;
+  nativeCurrency?: {
+    name: string;
+    symbol: string;
+    decimals: number;
+  };
+  rpcUrl?: string;
+}
+```
+
+Only the `chainId` is required per this specification, but a wallet **MAY** require any other fields listed, and impose additional requirements on them.
+
+- `chainId`
+  - **MUST** specify the integer ID of the chain as a hexadecimal string, per the [`eth_chainId`](./eip-695.md) Ethereum RPC method.
+- `blockExplorerUrl`
+  - If provided, **MUST** specify the URL of a block explorer web site for the chain.
+- `chainName`
+  - If provided, **MUST** specify a human-readable name for the chain.
+- `nativeCurrency`
+  - If provided, **MUST** describe the native currency of the chain using the `name`, `symbol`, and `decimals` fields.
+- `rpcUrl`
+  - If provided, **MUST** specify the URL of the RPC endpoint used to communicate with the chain.
+
+All URL string fields **MUST** include the protocol component of the URL.
+If the protocol of the URL is HTTP, HTTPS **SHOULD** be used.
```

Comment:
```suggestion
HTTPS **SHOULD** be used.
```
I'm not sure if this is what you meant or not, but it definitely seems wrong to say that "if HTTP is explicitly supplied, ignore it and use HTTPS instead".  If that is really what you mean, why not instead say that `HTTPS **MUST** be used if available, even if the URL provided is HTTP` or something?